### PR TITLE
OAPE-18: Bump openshift/api to vendor machineNamePrefix field

### PR DESCRIPTION
**Pull request by chiragkyal**
Bump `openshift/api` to vendor `machineNamePrefix` field and `CPMSMachineNamePrefix` feature-gate

- Part of : [OAPE-18](https://issues.redhat.com/browse/OAPE-18)
- Implementes : https://github.com/openshift/enhancements/pull/1714

### CODE REVIEW

**Comment by damdo** [comment]
The failure in units look legit @chiragkyal

**Comment by chiragkyal** [comment]
> The failure in units look legit @chiragkyal

Yeah, it's due to the `format` library for CEL validation. I asked @JoelSpeed about this in [Slack](https://redhat-internal.slack.com/archives/CE4L0F143/p1733137637295489?thread_ts=1731414877.498379&cid=CE4L0F143). Let me update the `ENVTEST_VERSION`

**Comment by chiragkyal** [comment]
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-control-plane-machine-set-operator/333/pull-ci-openshift-cluster-control-plane-machine-set-operator-main-unit/1863634503809896448#1:build-log.txt%3A26
```
unable to fetch hash for requested version: unable to find archive for 1.31.2 (linux,amd64)
exit status 2
KUBEBUILDER_ASSETS="" ./hack/test.sh
```

Looks like `1.31.2` binaries are not available in upstream yet.

```
go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
go: downloading sigs.k8s.io/controller-runtime v0.19.3
----
setup-envtest use
Version: 1.31.0
```

**Comment by damdo** [comment]
Yeah, let's use 1.31.0 then.
Thanks

File: Makefile [hunk 1]
```diff
@@ -91,7 +91,7 @@ test: generate fmt vet unit ## Run tests.
 
 .PHONY: unit
 unit: ## Run only the tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path --bin-dir $(PROJECT_DIR)/bin)" ./hack/test.sh
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path --bin-dir $(PROJECT_DIR)/bin --index https://raw.githubusercontent.com/openshift/api/master/envtest-releases.yaml)" ./hack/test.sh
```
Comment (chiragkyal) [inline Makefile:94]: We can change the index reference to this repository in a follow up PR

**Comment by chiragkyal** [comment]
Looks like I'm hitting [an issue](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-control-plane-machine-set-operator/333/pull-ci-openshift-cluster-control-plane-machine-set-operator-main-unit/1863898072707239936#1:build-log.txt%3A7501-7534) related to https://github.com/openshift/api/pull/2074 PR. 

```
              Status: "Failure",
              Message: "admission webhook \"controlplanemachineset.machine.openshift.io\" denied the request: spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace: Internal error: workspace fields should not be set when control plane nodes are in a failure domain: &v1beta1.Workspace{Server:\"test-vcenter\", Datacenter:\"\", Folder:\"\", Datastore:\"\", ResourcePool:\"\", VMGroup:\"\"}",
              Reason: "Forbidden",
              Details: nil,
              Code: 403,
          },
      }
  to match error
      <*matchers.AndMatcher | 0xc000ab4fc0>: {
          Matchers: [
              <*matchers.EqualMatcher | 0xc0009004a0>{
                  Expected: <string>"admission webhook \"controlplanemachineset.machine.openshift.io\" denied the request: spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace: Internal error: workspace fields should not be set when control plane nodes are in a failure domain: &v1beta1.Workspace{Server:\"test-vcenter\", Datacenter:\"\", Folder:\"\", Datastore:\"\", ResourcePool:\"\"}",
              },
          ],
```

@jcpowermac Could you please help fix the issue?

File: envtest-releases.yaml
Comment (JoelSpeed) [inline envtest-releases.yaml:1]: Index should only live in the o/api repo, please don't commit this here

File: Makefile (hunk 1 above)
Comment (JoelSpeed) [reply Makefile:94]: No, it should stay pointing at o/api please, keeping that as a central openshift source of truth

File: envtest-releases.yaml
Comment (chiragkyal) [reply envtest-releases.yaml:1]: Sure sounds good, I've removed the file.

File: Makefile (hunk 1 above)
Comment (chiragkyal) [reply Makefile:94]: Sure, thanks for the suggestion.

**Comment by jcpowermac** [comment]
> Looks like I'm hitting [an issue](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-control-plane-machine-set-operator/333/pull-ci-openshift-cluster-control-plane-machine-set-operator-main-unit/1863898072707239936#1:build-log.txt%3A7501-7534) related to [openshift/api#2074](https://github.com/openshift/api/pull/2074) PR.
> 
> ```
>               Status: "Failure",
>               Message: "admission webhook \"controlplanemachineset.machine.openshift.io\" denied the request: spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace: Internal error: workspace fields should not be set when control plane nodes are in a failure domain: &v1beta1.Workspace{Server:\"test-vcenter\", Datacenter:\"\", Folder:\"\", Datastore:\"\", ResourcePool:\"\", VMGroup:\"\"}",
>               Reason: "Forbidden",
>               Details: nil,
>               Code: 403,
>           },
>       }
>   to match error
>       <*matchers.AndMatcher | 0xc000ab4fc0>: {
>           Matchers: [
>               <*matchers.EqualMatcher | 0xc0009004a0>{
>                   Expected: <string>"admission webhook \"controlplanemachineset.machine.openshift.io\" denied the request: spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace: Internal error: workspace fields should not be set when control plane nodes are in a failure domain: &v1beta1.Workspace{Server:\"test-vcenter\", Datacenter:\"\", Folder:\"\", Datastore:\"\", ResourcePool:\"\"}",
>               },
>           ],
> ```
> 
> @jcpowermac Could you please help fix the issue?

see #325

**Comment by chiragkyal** [comment]
> see https://github.com/openshift/cluster-control-plane-machine-set-operator/pull/325

Thanks. I hope https://github.com/openshift/cluster-control-plane-machine-set-operator/pull/333/commits/bec86f9898d718d8af73eea7707a07c10472ad93 will make the unit tests happy for now.

**Comment by chiragkyal** [comment]
/assign @JoelSpeed @damdo 
up for approval

**Review by damdo — COMMENTED** [review]
/lgtm

/retest-required

**Comment by chiragkyal** [comment]
@damdo, could you help in approval as well? Thanks!

**Review by damdo — COMMENTED** [review]
/approve

It looks reasonable to me.

**Pull request merged**